src/lib/Server/Output/ValueObjectVisitor/RestContentType.php

```diff
@@ -154,6 +154,15 @@ public function visit(Visitor $visitor, Generator $generator, $data)
             );
         }
 
+        $imageFields = $contentType->getFieldDefinitions()->filterByType('ezimage');
+        $generator->startHashElement('imageFields');
+        $generator->startList('identifier');
+        foreach ($imageFields as $imageField) {
+            $generator->valueElement('value', $imageField->identifier);
+        }
+        $generator->endList('identifier');
+        $generator->endHashElement('imageFields');
```

Comment:
This violates Open-Closed principle and causes maintenance issues. Imagine you need to add handling of another Field Type. And then another. And another. You would need to create spaghetti code here. What about custom Field Types? :)

Reply:
I've looked at a few options and the only correct option in my opinion will be the number 2 you provided. Number 3, in my opinion, is too radical, because this functionality is not used all the time, and downloading and filtering each CT with each refresh is pointless (some of our clients have very extensive CTs)